internal/hash/checksum.go

```diff
@@ -240,6 +240,50 @@ func ReadCheckSums(b []byte, part int) map[string]string {
 	return res
 }
 
+// ReadPartCheckSums will read all part checksums from b and return them.
+func ReadPartCheckSums(b []byte) []map[string]string {
+	var res []map[string]string
+	for len(b) > 0 {
+		t, n := binary.Uvarint(b)
+		if n < 0 {
+			break
```

Comment:
I think, let's return an error here instead ?

Reply:
No real need. We don't need errors, instead, like the rest it just doesn't add the checksums.

Reply:
oh yes, sorry I did not mean in this particular line; but in general, it is better to log when we have unexpected data, so it will be easier to debug when something bad happens. Wouldn't that be the case here ?

Reply:
I'm pretty new to the checksum part of our codebase but I'll talk with @klauspost about where/when we could place error logs, assuming they make sense. 

Replying to comment below from @vadmeste 
```
oh yes, sorry I did not mean in this particular line; but in general, it is better to log when we have unexpected data, so it will be easier to debug when something bad happens. Wouldn't that be the case here ? 
```

Reply:
It is kinda non-actionable and not really important...

It is not worth failing over... And users will just be confused about the output.